### PR TITLE
30480 Alert message messed up for urgent care and no open slots

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
The PR fixes the alert message for urgent care when there are slots and no slots

## Original issue(s)
department-of-veterans-affairs/va.gov-team#30480

## Testing done
Unit testing

## Screenshots
localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_select-date (1)
---
localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_select-date (2)

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by grossb1 — APPROVED** [review]
Looks good to me.

**Review by jenniemc — COMMENTED** [review]
Hi @vbahinwillit , can you include a screenshot of what the alert box would look like when it's working as expected? And how did you set up the files to duplicate the bug so I can do the same locally?

**Comment by vbahinwillit** [comment]
> Hi @vbahinwillit , can you include a screenshot of what the alert box would look like when it's working as expected? And how did you set up the files to duplicate the bug so I can do the same locally?

To test, you can modify ./vaos/services/mocks/v2/slots.json to simulate no available slots. Set the preferred appointment date to the current date (urgent care) in the app.

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx [hunk 1]
```diff
@@ -33,7 +82,7 @@ export const WaitTimeAlert = ({
   if (momentPreferredDate.isValid() && momentNextAvailableDate.isValid()) {
     const showUrgentCareMessage = today.isSame(momentPreferredDate, 'day');
 
-    if (showUrgentCareMessage) {
+    if (showUrgentCareMessage && nextAvailableApptDate) {
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx:85]: Add a boolean to replace the variable string `nextAvailableApptDate`
```suggestion
  const hasNextAvailableApptDate = !!nextAvailableApptDate;
  if (showUrgentCareMessage && hasNextAvailableApptDate) {

```

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx [hunk 2]
```diff
@@ -60,45 +109,37 @@ export const WaitTimeAlert = ({
...
+            <UrgentCareMessage />
+            <ActionButtons
+              eligibleForRequests={eligibleForRequests}
+              facilityId={facilityId}
+              onClickRequest={onClickRequest}
+            />
+          </>
+        </InfoAlert>
+      );
+    } else {
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx:121]: This specifies for no slots available plus will account for both same day (urgent care) and non-same day appts.
```suggestion
    } else if (!hasNextAvailableApptDate) {
```

**Review by jenniemc — CHANGES_REQUESTED** [review]
Just some minor tweaks to the logic so if there are future dates available for urgent care (same day request), it uses "The earliest we can schedule..." alert box instead of "We couldn't find an appointment."

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx:85]: Done.

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/DateTimeSelectPage/WaitTimeAlert.jsx:121]: Good catch! Done.

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**